cpp/dynamic_community_detection_module/dynamic_community_detection_module.cpp

```diff
@@ -0,0 +1,347 @@
+constexpr std::string_view const kWeighted{"weighted"};
+constexpr std::string_view const kSimilarityThreshold{"similarity_threshold"};
+constexpr std::string_view const kExponent{"exponent"};
+constexpr std::string_view const kMinValue{"min_value"};
+constexpr std::string_view const kWeightProperty{"weight_property"};
+constexpr std::string_view const kWSelfloop{"w_selfloop"};
+constexpr std::string_view const kMaxIterations{"max_iterations"};
+constexpr std::string_view const kMaxUpdates{"max_updates"};
+
+constexpr std::string_view const kCreatedVertices{"createdVertices"};
+constexpr std::string_view const kCreatedEdges{"createdEdges"};
+constexpr std::string_view const kUpdatedVertices{"updatedVertices"};
+constexpr std::string_view const kUpdatedEdges{"updatedEdges"};
+constexpr std::string_view const kDeletedVertices{"deletedVertices"};
+constexpr std::string_view const kDeletedEdges{"deletedEdges"};
+
+namespace {
+LabelRankT::LabelRankT algorithm = LabelRankT::LabelRankT();
+bool initialized = false;
+
+auto saved_directedness = false;
+auto saved_weightedness = false;
+std::string saved_weight_property = "weight";
+
+constexpr double DEFAULT_WEIGHT = 1.0;
+}  // namespace
+
+void InsertCommunityDetectionRecord(mgp_graph *graph, mgp_result *result,
+                                    mgp_memory *memory, std::uint64_t node_id,
+                                    std::uint64_t community_id) {
+  auto *record = mgp::result_new_record(result);
+
+  mg_utility::InsertNodeValueResult(graph, record, kFieldNode.data(), node_id,
+                                    memory);
+  mg_utility::InsertIntValueResult(record, kFieldCommunityId.data(),
+                                   community_id, memory);
+}
+
+void InsertMessageRecord(mgp_result *result, mgp_memory *memory,
+                         const char *message) {
+  auto *record = mgp::result_new_record(result);
+
+  mg_utility::InsertStringValueResult(record, kFieldMessage.data(), message,
+                                      memory);
+}
+
+void Set(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result,
+         mgp_memory *memory) {
+  try {
+    const auto directed = mgp::value_get_bool(mgp::list_at(args, 0));
+    const auto weighted = mgp::value_get_bool(mgp::list_at(args, 1));
+    const auto similarity_threshold =
+        mgp::value_get_double(mgp::list_at(args, 2));
+    const auto exponent = mgp::value_get_double(mgp::list_at(args, 3));
+    const auto min_value = mgp::value_get_double(mgp::list_at(args, 4));
+    const auto weight_property = mgp::value_get_string(mgp::list_at(args, 5));
+    const auto w_selfloop =
+        weighted ? mgp::value_get_double(mgp::list_at(args, 6)) : 1.0;
+    const auto max_iterations = mgp::value_get_int(mgp::list_at(args, 7));
+    const auto max_updates = mgp::value_get_int(mgp::list_at(args, 8));
+
+    ::saved_directedness = directed;
+    ::saved_weightedness = weighted;
+    ::saved_weight_property = weight_property;
```

Comment:
```suggestion
    saved_directedness = directed;
    saved_weightedness = weighted;
    saved_weight_property = weight_property;
```

Reply:
In past discussion some reviewers didn’t find it immediately evident that these variables were global so I’ve elected to use the scope resolution operator. As it doesn’t do any harm, I’ll keep it there for now unless there’s a reason not to use it.

Reply:
In past discussion some reviewers didn’t find it immediately apparent that the left-hand variables were global so I’ve elected to use the scope resolution operator. As it doesn’t do any harm, I would prefer to keep it here for added clarity.